src/control/actor/host/notify.go

```diff
@@ -60,14 +60,28 @@ func (c *HostNotifyCmd) connectedF(net network.Network, conn network.Conn) {
 
 	// request metadata as soon as we connect to a peer
 	peerData, err := PollPeerMetadata(conn.RemotePeer(), c.Base, c.PeerMetadataState, c.Store, c.PeerStore)
+	// request BeaconStatus metadata as we connect to a peer
+	h, err := c.Host()
+	if err != nil {
+		log.Error(err)
+	}
+	bState, err2 := nodemetadata.ReqBeaconStatus(context.Background(), h, conn.RemotePeer())
```

Comment:
you can also use `err`

Reply:
Be careful when you continue after an error. Lets say that `h, err := c.Host()` errors. In that case you wouldn't want to continue calling `ReqBeaconStatus` because the content of `h` will be most likely `nil`. So most likely you will get a nul pointer exception here when line 64 errors. The whole point of errors is not continuing if there was one, as nasty things can happen.

Reply:
there is actually not much sense in tracking that error, if there is no host, it is impossible to get notified from a message and arrive at that function. I will dismiss the error making the sequence cleaner.